pkg/app/api/grpcapi/web_api.go

```diff
@@ -643,6 +643,8 @@ func (a *WebAPI) ListUnregisteredApplications(ctx context.Context, _ *webservice
 		return &webservice.ListUnregisteredApplicationsResponse{}, nil
 	}
 
+	// TODO: Eliminate apps registered by another Piped
+	//   In case that multiple Pipeds watches the same repo, it could be happen that one Piped report App-1 as an unregistered app even though another Piped register that app.
 	sort.Slice(allApps, func(i, j int) bool {
 		return allApps[i].Path < allApps[j].Path
 	})
```

Comment:
This TODO and this `sort` depend on how the UI looks like.
So I think we should fix the UI before continuing on updating this web API.

For example, in case the UI flow is as below, this API will receive a Piped ID and then return just ones belonging to that Piped.

- User picks where they want to deploy first (select one Piped and one cloud provider)
- Then API will be called to show the list of suggested apps for that Piped and cloud provider
- User picks app they want to create

Reply:
It does 👍 Let me drop this one for now.

Reply:
Dropped 👍 1c84d7b